web/text_layer_builder.js

```diff
@@ -187,107 +187,112 @@ class TextLayerBuilder {
     this.#selectionChangeAbortController = new AbortController();
     const { signal } = this.#selectionChangeAbortController;
 
+    const on = (event, handler) =>
+      document.addEventListener(event, handler, { signal });
+
     const reset = (end, textLayer) => {
       if (typeof PDFJSDev === "undefined" || !PDFJSDev.test("MOZCENTRAL")) {
         textLayer.append(end);
         end.style.width = "";
         end.style.height = "";
       }
-      end.classList.remove("active");
+      textLayer.classList.remove("selecting");
     };
 
-    document.addEventListener(
-      "pointerup",
-      () => {
+    let isPointerDown = false;
+    on("pointerdown", () => {
+      isPointerDown = true;
+    });
+    on("pointerup", () => {
+      isPointerDown = false;
+      this.#textLayers.forEach(reset);
+    });
+    on("keyup", () => {
+      if (!isPointerDown) {
         this.#textLayers.forEach(reset);
-      },
-      { signal }
-    );
+      }
+    });
```

Comment:
Don't we need a "blur" listener as well, such that the state is reset correctly if user leaves the browser window since that may prevent pointerup/keyup events from being received?

Reply:
Good point, let me try

Reply:
Yes we indeed need it. Browsers still fire "pointerup" when a user starts selecting and moves the mouse outside of the window, but as soon as somebody clicks (with a different mouse button) outside of the window then pointerup is not fired.